const.md

```diff
@@ -23,31 +26,41 @@ const REF: &u32 = { const _VAL = EXPR; static _STATIC = EXPR; &_STATIC };
 (`EXPR` is assigned to a `const` first to make it subject to the restrictions
 discussed in this document.)
 
-There are three reasons why this could be an issue.
+There are various reasons why this could be an issue.
 
-### Pointer equality
+### 1. Pointer equality
 
 We effectively "deduplicate" all the allocations that would otherwise locally be
 created at each use site of `REF`.  This is observable when the program compares
 these pointers for equality.  We consider this okay, i.e., programs may not rely
 on such constants all getting distinct addresses.  They may not rely on them all
 getting the same address either.
 
-### Interior mutability
+### 2. Interior mutability
 
 If the reference has type `&Cell<i32>` it is quite clear that the program can
 easily observe whether two references point to the same memory even without
 comparing their address: Changes through one reference will affect reads through
 the other.  So, we cannot allow constant references to types that have interior
-mutability (types that are not `Freeze`).
+mutability (types that are not `Freeze`):
+
+```rust
+const BAD: &Cell<i32> = &Cell::new(42);
+// Inlining `BAD` everywhere clearly is not the same as them all pointing to the same thing.
+```
 
 However, we can do better than that: Even if a *type* is not `Freeze`, it can
 have *values* that do not exhibit any interior mutability.  For example, `&None`
 at type `&Option<Cell<i32>>` would be rejected by the naive analysis above, but
 is actually accepted by the compiler because we know that there is no
 `UnsafeCell` here that would permit interior mutability.
 
-### `Sync`
+*Dynamic check.* The Miri engine could check this dynamically by ensuring that
```

Comment:
I thought we were doing this nowadays?

Reply:
Not that I know of. I did not have any time recently to work on https://github.com/rust-lang/rust/issues/49206.

Reply:
we have https://github.com/rust-lang/rust/blob/0444b9f66acb5da23dc816e0d8eb59623ba9ea50/src/librustc_mir/interpret/intern.rs#L183

Reply:
Oh sorry, I saw the `Sync` and thought this was about `Sync`. But that's a red line in the diff.

Yes, I think we might be already doing this, but I'd like to review the code specifically checking for this (and maybe add some comments saying why this is important) before making claims here.